components/cam/src/physics/cam/debug_info.F90

```diff
@@ -0,0 +1,81 @@
+module debug_info
+
+  !-------------------------------------------------------------
+  ! This is a light weight module to report parameters which
+  ! might be of interest in case of a simulation crash.
+  !
+  ! Author: Balwinder Singh
+  ! (Following suggestions from Peter Caldwell and Aaron Donahue)
+  !-------------------------------------------------------------
+
+  implicit none
+  private
+
+  !Variables updated during the simulation
+  integer :: macmiciter, chunk
+  !$OMP THREADPRIVATE(macmiciter,chunk )
+
+  public:: &
+       get_debug_chunk, &       !update chunk info from the simulation
+       get_debug_macmiciter, &  !update macro/microphysics iteration count
+       report_error_info        !report info on the current lat/lon and other simulation parameters
+
+contains
+
+  subroutine get_debug_chunk(chunk_in)
+
+    !Obtain chunk # from tphycbc
+
+    implicit none
+
+    integer :: chunk_in
+
+    chunk = chunk_in
+
+  end subroutine get_debug_chunk
+
+
+  subroutine get_debug_macmiciter(iter)
+
+    !Obtain iteration # of macro/microphysics sub-stepping iteration
+
+    implicit none
+
+    integer, intent(in) :: iter
+
+    macmiciter = iter
+
+  end subroutine get_debug_macmiciter
+
+
+  subroutine report_error_info(cause, calling_func, icol, klev)
+
+    !Reports simulation parameters
+    use cam_logfile,    only: iulog
+    use phys_grid,      only: get_rlat_p, get_rlon_p
+    use time_manager,   only: get_nstep
+    use physics_utils,  only: rtype
+    implicit none
+
+    integer, intent(in) :: icol !column(lat/lon) index
+    integer, optional, intent(in) :: klev !vertical index
+    character(len=*), intent(in) :: cause, calling_func
+
+    !local variables
+    real(rtype), parameter::  radian2deg = 180.0_rtype/(4.0_rtype*atan(1.0_rtype)) !180/pi = 57.296
+
+    write(iulog,'(a)') '*****************************************************************'
```

Comment:
Quick question, if multiple PE's report an error would we expect the following lines to be scrambled?  Meaning, multiple instances of writing to e3sm.log?  The masterproc would write to the atm.log file but any other procs could compete to write to e3sm.log.  Would it be possible/prudent to include rank info in the message?

This is really just a question, I'm happy to keep this as is.  But I thought I would bring it up.  Since this is a message we intend to have written only once and when there is an error it's not as big a deal as diagnostic output from multiple PE's could be.

Reply:
I think F90 won't scramble within lines but may interleave lines from different processors. In that case, a quick fix would be to have a single write command with lines separated by \n. Balwinder probably has a better grasp of the situation than I do though - perhaps I'm wrong.

Reply:
I think you are right, I meant to say that the lines will be interweaved.  On machines like Cori which attach the proc rank to every output line it shouldn't be an issue.  For machines like quartz or syrah it may mean that we have a bunch of lines that are difficult to parse.  All-in-all I posed as a question and not a requested change because if it takes too much time to implement a fix it's probably not worth the effort.  I'm happy to keep it as is.

Reply:
I think Peter is right. These write statements should be condensed into one with calls to `NEW_LINE` to separate lines. I think you are guaranteed atomic access to stdout within each call to write.

That said, since we won't know in which order the ranks print, it'd be nice to add the MPI rank to the message to be printed.

Reply:
Thanks! These are good points. I will change these multiple write statements into one with `new_line`. I separated them out to make it easy to use format statement so that they appear in a nice readable format.
As Aaron mentioned, E3SM log prints out the rank which is printing out a statement, for example:
```
 7: Fail due to CAUSE in subroutine/function:p3_main
 7: Column location in degrees(lat,lon) is:(-38.61614714,353.78115295)
 7: macmic_it is:    1
 7: timestep:    0
 7: K level is:   72

```
`7:` here is the rank which printed these statements. If there are machines which doesn't print this info, it is fairly easy to get hold of it using `iam` variable from `spmd_utils`